### PR TITLE
Make home directory location setting optional

### LINKED ISSUE #559: New Home Attribute on consul user resource breaks our puppet runs.
This merge https://github.com/solarkennedy/puppet-consul/pull/550 breaks our Puppet runs on our current Consul Agents.

```
Error: Could not set home on user[consul]: Execution of '/usr/sbin/usermod -d /opt/consul/data consul' returned 8: usermod: user consul is currently used by process 894
Error: /Stage[main]/Consul::Install/User[consul]/home: change from '/home/consul' to '/opt/consul/data' failed: Could not set home on user[consul]: Execution of '/usr/sbin/usermod -d /opt/consul/data consul' returned 8: usermod: user consul is currently used by process 894
```

As far as I can see, many popular Puppet Modules avoid the User resource, or when they use them, they do not set the home attribute.

How could this be resolved cleanly?

### PR DESCRIPTION
**Pull request by genebean**
Fixes #559

### CODE REVIEW

**Comment by genebean** [comment]
@bastelfreak I am not sure I picked the right label here - feel free to change if I goofed

**Review by bastelfreak — APPROVED** [review]
I did some chatting with @genebean on IRC and I think  this is the best approach to move forward. It worka for new and existing environments and in my opinion that is important.

**Comment by genebean** [comment]
@bastelfreak how do you feel about merging this and cutting a release?

File: manifests/init.pp [hunk 1]
```diff
@@ -117,6 +117,12 @@
 # [*manage_user*]
 #   Whether to create/manage the user that should own consul's configuration files.
 #
+# [*manage_user_home_location*]
+#   Whether to explicitly set the location of teh consul user's home directory when this modules
```
Comment (solarkennedy) [inline manifests/init.pp:121]: `teh`

File: manifests/init.pp [hunk 2]
```diff
@@ -117,6 +117,12 @@
...
+#   manages the creation of the user. If the consul user already exists and this is enabled, puppet
```
Comment (solarkennedy) [inline manifests/init.pp:122]: Can you also note that this setting only takes effect if `manage_user` is true ?

**Review by solarkennedy — APPROVED** [review]

File: manifests/init.pp (hunk 1 above)
Comment (genebean) [reply manifests/init.pp]: fixed

File: manifests/init.pp (hunk 2 above)
Comment (genebean) [reply manifests/init.pp]: How does that look @solarkennedy ?

**Pull request merged**